inabox/tests/setup_infra.go

```diff
@@ -0,0 +1,211 @@
+package integration_test
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/Layr-Labs/eigenda/inabox/deploy"
+	"github.com/Layr-Labs/eigenda/test/testbed"
+	"github.com/Layr-Labs/eigensdk-go/logging"
+	"github.com/testcontainers/testcontainers-go/network"
+)
+
+// SetupGlobalInfrastructure creates the shared infrastructure that persists across all tests.
+// This includes containers for Anvil, LocalStack, GraphNode, and the Churner server.
+// This should be called once in TestMain.
+func SetupGlobalInfrastructure(
+	templateName, testName string,
+	inMemoryBlobStore bool,
+	logger logging.Logger,
+) (*InfrastructureHarness, error) {
+	infra := &InfrastructureHarness{
+		TemplateName:        templateName,
+		TestName:            testName,
+		InMemoryBlobStore:   inMemoryBlobStore,
+		Logger:              logger,
+		MetadataTableName:   "test-BlobMetadata",
+		BucketTableName:     "test-BucketStore",
+		MetadataTableNameV2: "test-BlobMetadata-v2",
+	}
+
+	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Minute)
```

Comment:
should probably have this injected instead of using random 5 minutes. tests might end up taking longer than 5 minutes at some point. we should just use t.Context() for tests.

Reply:
I can't use t.Context() here because it's part of TestMain

Reply:
I see. I was reading quickly. Keeping the hardcoded 5 mins is prob fine then given that this function would know better how long it takes to spin up the infra then its callers. Should prob still take a ctx as input to the function instead of using a context.Background() like this